gatsby-config.ts

```diff
@@ -9,12 +9,41 @@ export const plugins = [
   'gatsby-plugin-postcss',
   'gatsby-plugin-styled-components',
   'gatsby-plugin-image',
-  'gatsby-plugin-sitemap',
   'gatsby-plugin-sharp',
   'gatsby-transformer-yaml',
   'gatsby-transformer-sharp',
   'gatsby-plugin-react-helmet',
   'gatsby-plugin-ts',
+  {
+    resolve: 'gatsby-plugin-sitemap',
+    options: {
+      resolveSiteUrl: () => siteMetadata.siteUrl,
+      excludes: ['/_*'],
+      output: '/docs',
+      filterPages: ({ path }) => {
+        // NOTE: these didn't work in `exclude` field, so we had to use custom regex in filterPage function
+        if (
+          // Legacy versions of pages are excluded from the sitemap (anything with a URL containing .../versions/vx.x/where x is at least one digit, e.g. v1.1, v1.2
```

Comment:
I love this, you are doing every single thing in your power to make this code readable for the next person. Sometimes i make developers turn these comments into methods but I feel like that would make this harder to understand and it is SO easy to understand now.